ytypes/binary_type.go

```diff
@@ -35,12 +37,13 @@ func validateBinary(schema *yang.Entry, value interface{}) error {
 	}
 
 	// Check that type of value is the type expected from the schema.
-	binaryVal, ok := value.([]byte)
-	if !ok {
+	t := reflect.TypeOf(value)
```

Comment:
We should probably create a helper here -- `isBinaryField(reflect.Type)` and re-write this:

```golang
if t := reflect.TypeOf(value); !isBinaryType(t) { ...
```

Since we use the same logic below.

Reply:
Refactored logic to dedicated functions for:
* `isBinaryType` - to do type test previously handled by `.([]byte)`
* `isBinarySliceType` - to do type test previously handled by `.([][]byte)`